src/test/java/uk/gov/hmcts/juror/api/moj/report/grouped/UnconfirmedAttendanceReportTest.java

```diff
@@ -0,0 +1,160 @@
+        return new UnconfirmedAttendanceReport(this.courtLocationService);
+    }
+
+    @Override
+    protected StandardReportRequest getValidRequest() {
+        return StandardReportRequest.builder()
+            .reportType("UnconfirmedAttendanceReport")
+            .fromDate(LocalDate.of(2024, 1, 1))
+            .toDate(LocalDate.of(2024, 1, 2))
+            .build();
+    }
+
+    @Override
+    public void positivePreProcessQueryTypical(JPAQuery<Tuple> query, StandardReportRequest request) {
+        request.setFromDate(LocalDate.of(2024, 1, 1));
+        request.setToDate(LocalDate.of(2024, 1, 2));
+        securityUtilMockedStatic.when(SecurityUtil::getLocCode).thenReturn(TestConstants.VALID_COURT_LOCATION);
+        report.preProcessQuery(query, request);
+
+        verify(query).where(QAppearance.appearance.attendanceDate.between(request.getFromDate(), request.getToDate())
+            .and(QAppearance.appearance.appearanceStage.in(
+                AppearanceStage.CHECKED_IN,
+                AppearanceStage.CHECKED_OUT
+            )));
+        verify(query).where(QAppearance.appearance.locCode.eq(SecurityUtil.getLocCode()));
+        verify(query).orderBy(QAppearance.appearance.attendanceDate.desc(), QAppearance.appearance.jurorNumber.asc());
+    }
+
+    @Override
+    public Map<String, AbstractReportResponse.DataTypeValue> positiveGetHeadingsTypical(
+        StandardReportRequest request,
+        AbstractReportResponse.TableData<GroupedTableData> tableData,
+        GroupedTableData data) {
+
+        request.setFromDate(LocalDate.of(2024, 1, 1));
+        request.setToDate(LocalDate.of(2024, 1, 2));
+        securityUtilMockedStatic.when(SecurityUtil::getLocCode).thenReturn(TestConstants.VALID_COURT_LOCATION);
+
+        when(data.getSize()).thenReturn(50);
+
+        CourtLocation courtLocation = mock(CourtLocation.class);
+        when(courtLocationService.getCourtLocation(TestConstants.VALID_COURT_LOCATION)).thenReturn(courtLocation);
+        doReturn(getCourtNameEntry()).when(report).getCourtNameHeader(courtLocation);
+
+        Map<String, GroupedReportResponse.DataTypeValue> expected = new ConcurrentHashMap<>();
+        expected.put("total_unconfirmed_attendances", GroupedReportResponse.DataTypeValue.builder()
+             .displayName("Total unconfirmed attendances")
+             .dataType(Long.class.getSimpleName())
+             .value(tableData.getData().getSize())
+             .build());
+        expected.put("court_name", getCourtNameEntry().getValue());
+
+        Map<String, StandardReportResponse.DataTypeValue> map = report.getHeadings(request, tableData);
+        assertHeadingContains(map,
+            request,
+            false,
+            expected
+        );
+
+        verify(report).getHeadings(request, tableData);
+        verify(report, times(1)).getCourtNameHeader(any());
+        verifyNoMoreInteractions(report);
```

Comment:
Not needed as will be auto applied apart of the abstract test

Reply:
updated